src/adm.rs

```diff
@@ -1,43 +1,275 @@
+                // `default`.
+                result.position = filter.position.or(default.position);
+                Some(result)
+            }
+            None => {
+                self.report(
+                    &HandlerErrorKind::UnexpectedAdvertiser(tile.name).into(),
+                    tags,
+                );
+                None
+            }
+        }
+    }
+}
+
+/// Construct the AdmFilter from the provided settings.
+/// This uses a JSON construct of settings, e.g.
+/// ```javascript
+/// /* for the Example Co advertiser... */
+/// {"Example": {
+///     /* The allowed hosts for URLs */
+///     "advertiser_hosts": ["www.example.org", "example.org"],
+///     /* Valid tile positions for this advertiser (empty for "all") */
+///     "positions": [1, 2],
+///     /* Valid target regions for this advertiser
+///        (use "en-US" for "all in english speaking United States") */
+///     "include_regions": ["en-US/TX", "en-US/CA"],
+///     /* Allowed hosts for impression URLs.
+///        Empty means to use the impression URLs in "DEFAULT" */
+///     "impression_hosts: [],
+///     },
+///     ...,
+///  "DEFAULT": {
+///    /* The default impression URL host to check for. */
+///    "impression_hosts": ["example.net"]
+///     }
+/// }
+/// ```
+///
+impl From<&Settings> for HandlerResult<AdmFilter> {
+    fn from(settings: &Settings) -> Self {
+        let mut filter_map: BTreeMap<String, AdmAdvertiserFilterSettings> = BTreeMap::new();
+        for (adv, setting) in settings.adm_settings.clone() {
+            dbg!("Processing records for {:?}", &adv);
+            // map the settings to the URL we're going to be checking
+            filter_map.insert(adv.to_lowercase(), setting);
+        }
+        Ok(AdmFilter {
+            filter_set: filter_map,
+        })
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct AdmTile {
     pub id: u64,
     pub name: String,
     pub advertiser_url: String,
     pub click_url: String,
     pub image_url: String,
     pub impression_url: String,
+    pub position: Option<u8>,
 }
 
 pub async fn get_tiles(
     reqwest_client: &reqwest::Client,
     adm_endpoint_url: &str,
-    fake_ip: &str,
+    location: &LocationResult,
     stripped_ua: &str,
     placement: &str,
+    state: &ServerState,
+    tags: &mut Tags,
 ) -> Result<AdmTileResponse, HandlerError> {
     // XXX: Assumes adm_endpoint_url includes
     // ?partner=<mozilla_partner_name>&sub1=<mozilla_tag_id> (probably should
     // validate this on startup)
+    let settings = &state.settings;
     let adm_url = Url::parse_with_params(
         adm_endpoint_url,
         &[
-            ("ip", fake_ip),
-            ("ua", &stripped_ua),
-            ("sub2", &placement),
+            ("partner", settings.partner_id.as_str()),
+            ("sub1", settings.sub1.as_str()),
+            // ("sub2", placement),
+            ("country-code", &location.country()),
```

Comment:
Looks like you've taken care of #36 here too -- note that this will break the dev env until adM rolls out support for their new Tile API which is supposed to be mid may.

I'm checking if the dev is being used by the client team

Reply:
Just using mostly what was there.
I'll comment out `country` for now.

Reply:
Commenting out just country won't help, the integration still won't work (temporarily) unless we restore ip/ua. I haven't heard back from the client dev yet either, though.